Enforce c10-fullness for all ops

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49613 Remove backward and requires_grad from Autograd backend key
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* #49164 Remove codegen logic to support non-c10-full ops
* **#49619 Enforce c10-fullness for all ops**
* #49138 Making ops c10-full: list of optional tensors

This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

smessmer committed:
Enforce c10-fullness for all ops

This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

trip036 commented:
Amazing on the no failures yet. Absolutely great considering the degree of depth the entire platforms success rate is in equivalent to the software they haveSent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Facebook Community Bot <notifications@github.com> Date: 12/18/20  5:26 PM  (GMT-06:00) To: pytorch/pytorch <pytorch@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [pytorch/pytorch] Enforce c10-fullness for all ops (#49619) 

💊 CI failures summary and remediations

As of commit 3fd0568 (more details on the Dr. CI page):



💚 💚 Looks good so far! There are no failures yet. 💚 💚



This comment was automatically generated by Dr. CI (expand for details).Follow this link to opt-out of these comments for your Pull Requests.

Please report bugs/suggestions to the (internal) Dr. CI Users group.

This comment has been revised 2 times.





—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

trip036 commented:
Absolutely look forward to the final 4 lines of this part of one of the hardest and rate python power Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Facebook Community Bot <notifications@github.com> Date: 12/18/20  5:26 PM  (GMT-06:00) To: pytorch/pytorch <pytorch@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [pytorch/pytorch] Enforce c10-fullness for all ops (#49619) 

💊 CI failures summary and remediations

As of commit 3fd0568 (more details on the Dr. CI page):



💚 💚 Looks good so far! There are no failures yet. 💚 💚



This comment was automatically generated by Dr. CI (expand for details).Follow this link to opt-out of these comments for your Pull Requests.

Please report bugs/suggestions to the (internal) Dr. CI Users group.

This comment has been revised 2 times.





—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

smessmer committed:
Update on "Enforce c10-fullness for all ops"


This is a minimal-change PR that enforces that all operators are c10-full by making it the default.

This does not clean up any code yet, that will happen in PRs stacked on top. But this PR already ensures
that there are no non-c10-full ops left and there will be no non-c10-full ops introduced anymore.

Differential Revision: [D25650198](https://our.internmc.facebook.com/intern/diff/D25650198/)

[ghstack-poisoned]

bhosmer reviewed (approved):
looks like xla build breaks?
```
Jan 03 16:03:05 AtenXlaType function missed override: Tensor& _index_put_impl_(Tensor& self, TensorList indices, const Tensor& values, bool accumulate, bool unsafe); // _index_put_impl_(Tensor,TensorList,Tensor,bool,bool)->Tensor
Jan 03 16:03:05 AtenXlaType function missed override: Tensor index(const Tensor& self, TensorList indices); // index(Tensor,TensorList)->Tensor
Jan 03 16:03:05 AtenXlaType function missed override: Tensor& index_put_(Tensor& self, TensorList indices, const Tensor& values, bool accumulate); // index_put_(Tensor,TensorList,Tensor,bool)->Tensor
```
